docs/DafnyRef/UserGuide.md

```diff
@@ -947,6 +947,76 @@ implementation.
 - The current backend also assumes the use of C++17 in order to cleanly and
   performantly implement datatypes.
 
+### 24.9.8. Supported features by target language
+
+Some Dafny features are not supported by every target language.
+The table below shows which features are supported by each backend.
+An empty cell indicated that a feature is not supported,
+while an X indicates that it is.
+
+Note that this information is currently based on code inspection
+and is not yet guaranteed to be updated as features and backends are modified,
+but such a mechanism is in the works.
+
+| Feature | C# | JavaScript | Go | Java | Python | C++ |
+|-|-|-|-|-|-|-|
+| [Unbounded integers](#sec-numeric-types) |  X  |  X  |  X  |  X  |  X  |  |
+| [Real numbers](#sec-numeric-types) |  X  |  X  |  X  |  X  |  X  |  |
+| [Ordinals](#sec-ordinals) |  X  |  X  |  X  |  X  |  X  |  |
+| [Function values](#sec-arrow-subset-types) |  X  |  X  |  X  |  X  |  X  |  |
+| [Iterators](#sec-iterator-types) |  X  |  X  |  X  |  |  |  |
+| [Collections with trait element types](#sec-collection-types) |  X  |  X  |  X  |  X  |  X  |  |
+| [User-defined types with traits as type parameters](#sec-trait-types) |  X  |  X  |  X  |  |  X  |  X  |
+| [External module names with only underscores](#sec-extern-decls) |  X  |  X  |  |  X  |  X  |  X  |
+| [Co-inductive datatypes](#sec-co-inductive-datatypes) |  X  |  X  |  X  |  X  |  |  |
+| [Multisets](#sec-multisets) |  X  |  X  |  X  |  X  |  |  |
+| [Runtime type descriptors](#) |  X  |  X  |  X  |  X  |  |  |
+| [Multi-dimensional arrays](#sec-multi-dimensional-arrays) |  X  |  X  |  X  |  X  |  X  |  |
+| [Map comprehensions](#sec-map-comprehension-expression) |  X  |  X  |  X  |  X  |  X  |  |
+| [Traits](#sec-trait-types) |  X  |  X  |  X  |  X  |  X  |  |
+| [Let-such-that expressions](#sec-let-expression) |  X  |  X  |  X  |  X  |  X  |  |
+| [Non-native numeric newtypes](#sec-newtypes) |  X  |  X  |  X  |  X  |  X  |  |
+| [Method synthesis](#sec-synthesize-attr) |  X  |  |  |  |  |  |
+| [External classes](#sec-extern-decls) |  X  |  X  |  X  |  X  |  X  |  |
+| [Instantiating the `object` type](#sec-object-type) |  X  |  X  |  X  |  X  |  X  |  |
+| [`forall` statements that cannot be sequentialized](#sec-forall-statement)[^feature-note-1] |  X  |  X  |  X  |  X  |  |  |
```

Comment:
Ah yes, I see now that this is better so that footnote links are not broken when the set of footnote changes (which it would if the labels are sequentially numbered), good call.